modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1272,6 +1346,8 @@ private void dropTableLocally(long causalityToken, CatalogTableDescriptor tableD
 
             CompletableFuture<?>[] stopReplicaFutures = new CompletableFuture<?>[partitions];
```

Comment:
Seems that proposed changes are too complicated. Let's step back for a while and discuss (and design) an expected flow.

- Should we treat initial table creation as a rebalance from null assignments to pending-initial one? Or we should consider it as direct assignments.stable management? 
- Do we treat assignments.stable handling differently while it's the initial set or assignments.switch? Is it reasonable to split given handling into two methods or merge them within one?